### DISCUSSION TITLE
[CH6] 템플릿 리터럴 타입 강제 변환

### DISCUSSION THREAD

**Discussion by innerstella** [innerstella/ES2024#7] [Q&A] [answered]
템플릿 리터럴에서 표현식의 평가 결과가 문자열이 아니더라도 문자열로 타입이 강제 변환되어서 삽입되는데 이로 인해 생길 수 있는 문제가 뭐가 있을까?

**Comment by Lv1GoM** [comment, marked as answer]
템플릿 리터럴에서 표현식의 평가 결과가 문자열이 아니더라도 문자열로 타입이 강제로 변환되어서 삽입된다.
이로 인해 생길 수 있는 문제
- 의도치 않은 문자열 변환
표현식의 평가 결과가 숫자, 불리언, 객체 등의 다른 데이터 타입일 경우, 개발자의 의도와는 상관없이 강제로 문자열로 변환된다. 이는 개발자가 의도하지 않은 결과를 초래할 수 있다.
- 연산 결과의 예측 어려움
표현식 내에서 복잡한 연산이 수행되는 경우, 개발자가 예상하지 못한 문자열 변환으로 인해 결과를 예측하기 어려울 수 있다.
- 부작용 문제
표현식 내에서 부작용이 발생하는 경우, 예상치 못한 결과를 초래할 수 있다. 표현식 내에서 상태를 변경하거나 다른 연산을 수행하는 경우, 그 부작용이 문자열로 변환된 결과에도 영향을 미친다.
- 타입 불일치 문제
만약 표현식의 평가 결과가 문자열과 함께 사용될 때 예상하지 못한 타입 불일치 문제가 발생할 수 있다.

이러한 문제를 피하려면 템플릿 리터럴을 사용할 때 강제로 문자열 변환되는 부분에 대해 주의 깊게 고려하고, 필요한 경우 명시적으로 문자열로 변환하여 사용하는 것이 좋다. 예를 들어, 숫자를 문자열로 변환하려면 ```String()``` 함수나 ```.toString()``` 메서드를 사용할 수 있다.
